dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -194,6 +198,26 @@ private static Validator createValidator() {
         return validatorFactory.getValidator();
     }
 
+    /**
+     * Decide whether a gitReference is excluded, given a workflow/service's filters
+     * @param gitRefPath Path.of(gitReference) for glob matching with PathMatcher
+     * @param filters Filters specified for a workflow/service in .dockstore.yml
+     * @return
+     */
+    public static boolean filterGitReference(final Path gitRefPath, final List<String> filters) {
+        return filters.isEmpty() || filters.stream().anyMatch(filter -> {
+            String matcherString;
+            // Use regex if filter string is surrounded by /
+            if (filter.matches("^\\/.*\\/$")) {
```

Comment:
Edge case: What if the pattern is `//`? Will it match everything or nothing? Probably OK.

Reply:
It will match nothing.
If the pattern is `//` the effective regex will be just the prepended part `^refs\\/(heads\\/|tags\\/|)`.
That can only match strings `refs/`, `refs/heads/`, & `refs/tags/` which are all impossible (`Path` objects don't even keep the trailing `/`).